File: benchmarks/record_build.js
const { performance } = require('perf_hooks');

function build(n) {
    const s = { items: [], total: 0 };
    for (let i = 0; i < n; i++) {
        s.items.push(i);
        s.total += i;
    }
    return s;
}

const t0 = performance.now();
const s = build(500000);
console.log(s.total + s.items.length);
console.log(`bench_ns=${Math.round((performance.now() - t0) * 1e6)}`);

File: benchmarks/record_build.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)

class Stack
  attr_accessor :items, :total
  def initialize
    @items = []
    @total = 0
  end
end

def build(n)
  s = Stack.new
  i = 0
  while i < n
    s.items << i
    s.total += i
    i += 1
  end
  s
end

s = build(500000)
puts (s.total + s.items.length)
puts "bench_ns=#{Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond) - t0}"
